coordinator/src/main/scala/filodb.coordinator/FilodbCluster.scala

```diff
@@ -77,7 +77,8 @@ final class FilodbCluster(val system: ExtendedActorSystem, overrideConfig: Confi
     * All actions are idempotent. It manages the underlying lifecycle of all node actors.
     */
   private[coordinator] lazy val guardian = system.actorOf(NodeGuardian.props(
-    settings, metaStore, memStore, DefaultShardAssignmentStrategy), guardianName)
+    settings, metaStore, memStore, new K8sStatefulSetShardAssignmentStrategy(useHostNameToResolveShards =
+        settings.config.getBoolean("shard-manager.enable-k8s-stateful-shard-strategy"))), guardianName)
```

Comment:
Can we pass in  `K8sStatefulSetShardAssignmentStrategy` or `DefaultShardAssignmentStrategy` based on the config?

Reply:
Yes that should be possible but the strategy already uses the ``DefaultShardAssignmentStrategy`` as fallback? Do you still want to based on flag pass one over other?

Reply:
Changes made